tutorials/sphinx_tuto/tensordict_shapes.py

```diff
@@ -0,0 +1,117 @@
+# as the original one, but whose entries are views of the corresponding entries in the
+# original tensordict.
+
+chunks = tensordict.split([3, 1], dim=1)
+assert chunks[0].batch_size == torch.Size([3, 3])
+assert chunks[1].batch_size == torch.Size([3, 1])
+torch.testing.assert_allclose(chunks[0]["a"], tensordict["a"][:, :-1])
+
+
+##############################################################################
+# Stacking and concatenating
+# --------------------------
+#
+# ``TensorDict`` can be used in conjunction with ``torch.cat`` and ``torch.stack``.
+#
+# Stacking ``TensorDict``
+# ^^^^^^^^^^^^^^^^^^^^^^^
+# By default, stacking is done in a lazy fashion, returning a ``LazyStackedTensorDict``
+# object. In this case values are only stacked on-demand when they are accessed. This
+# in cases where you have a large ``TensorDict`` with many entries, and you don't need
+# to stack all of them.
+
+cloned_tensordict = tensordict.clone()
+# no stacking happens on the next line
+stacked_tensordict = torch.stack([tensordict, cloned_tensordict], dim=0)
+print(stacked_tensordict)
+
+##############################################################################
+# If we index a ``LazyStackedTensorDict`` we recover the original ``TensorDict``.
+
+assert stacked_tensordict[0] is tensordict
+assert stacked_tensordict[1] is cloned_tensordict
+
+##############################################################################
+# Accessing a key in the ``LazyStackedTensorDict`` results in those values being
+# stacked. If the key corresponds to a nested ``TensorDict`` then we will recover
+# another ``LaxyStackedTensorDict``.
+
+assert stacked_tensordict["a"].shape == torch.Size([2, 3, 4])
+
+###############################################################################
+# If we want to have a contiguous ``TensorDict``, we can call ``.to_tensordict()``
+# or ``.contiguous()``. It is recommended to perform this operation before
+# accessing the values of the stacked ``TensorDict`` so that the stacking operation is
+# not performed more often than is necessary.
+
+assert isinstance(stacked_tensordict.contiguous(), TensorDict)
+assert isinstance(stacked_tensordict.to_tensordict(), TensorDict)
+
+###############################################################################
+# Concatenating ``TensorDict``
+# ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+# Concatenation is not done lazily, instead calling ``torch.cat`` on a list of
+# ``TensorDict`` instances simply returns a ``TensorDict`` whose entries are the
+# concatenated entries of the elements of the list.
+
+concatenated_tensordict = torch.cat([tensordict, cloned_tensordict], dim=0)
+assert isinstance(concatenated_tensordict, TensorDict)
+assert concatenated_tensordict.batch_size == torch.Size([6, 4])
+assert concatenated_tensordict["b"].shape == torch.Size([6, 4, 5])
+
+##############################################################################
```

Comment:
Missing:
Unsqueeze, squeeze, permute, gather, indexing, setting items (if they won't be covered elsewhere)
For those where a dim is passed we should mention what behaviour is expected when using negative dim (eg unsqueeze(-1) adds one singleton after the last dim of the batch size, not the last of each tensor) and what will happen for nested tensordicts with more dims in that case (ie if a nested TD has 1 more dim than the root, unsqueeze(-1) should do an unsqueeze(-2) on it).

Finally we can also say that `apply` can be used for all the functions that are not covered.